data/pqtl/02.analyze-GpBPpG.ipynb

```diff
@@ -0,0 +1,2043 @@
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "repo_url = 'https://github.com/dhimmel/hetionet'\n",
+    "commit = '6d26d15e9055b33b4fd97a180fa288e4f2060b96'\n",
+    "names = ['hetionet-v1.0'] + [f'hetionet-v1.0-perm-{i + 1}' for i in range(5)]    \n",
+    "paths = ['hetnet/json/hetionet-v1.0.json.bz2'] + [\n",
+    "    f'hetnet/permuted/json/{name}.json.bz2' for name in names[1:]\n",
+    "]\n",
+    "hetnets = collections.OrderedDict()\n",
+    "for name, path in zip(names, paths):\n",
+    "    url = f'{repo_url}/raw/{commit}/{path}'\n",
+    "    hetnets[name] = hetio.readwrite.read_graph(url)\n",
+    "list(hetnets)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Computing DWPC matrix for the GpBPpG metapath in hetionet-v1.0 took 5.1 seconds\n",
+      "Computing DWPC matrix for the GpBPpG metapath in hetionet-v1.0-perm-1 took 6.6 seconds\n",
+      "Computing DWPC matrix for the GpBPpG metapath in hetionet-v1.0-perm-2 took 6.5 seconds\n",
+      "Computing DWPC matrix for the GpBPpG metapath in hetionet-v1.0-perm-3 took 6.6 seconds\n",
+      "Computing DWPC matrix for the GpBPpG metapath in hetionet-v1.0-perm-4 took 6.6 seconds\n",
+      "Computing DWPC matrix for the GpBPpG metapath in hetionet-v1.0-perm-5 took 6.5 seconds\n"
+     ]
+    }
+   ],
+   "source": [
+    "DWPCs = collections.OrderedDict()\n",
+    "for name, graph in hetnets.items():\n",
+    "    metapath = graph.metagraph.metapath_from_abbrev('GpBPpG')\n",
+    "    row_ids, col_ids, dwpc_matrix, seconds = hetmech.degree_weight.dwpc(graph, metapath, damping=0.4, dense_threshold=0.6)\n",
+    "    DWPCs[name] = dwpc_matrix\n",
+    "    print(f'Computing DWPC matrix for the {metapath} metapath in {name} took {seconds:.1f} seconds')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'Gene–participates–Biological Process–participates–Gene'"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "metapath.get_unicode_str()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Scaling as per https://think-lab.github.io/d/193/#4\n",
+    "dwpc_scaler = DWPCs['hetionet-v1.0'].mean()\n",
+    "for name in list(DWPCs):\n",
+    "    matrix = DWPCs[name]\n",
+    "    if scipy.sparse.issparse(matrix):\n",
+    "        DWPCs[name] = matrix.multiply(dwpc_scaler ** -1).arcsinh()\n",
+    "    else:\n",
+    "        DWPCs[name] = numpy.arcsinh(matrix / dwpc_scaler)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Unused code"
```

Comment:
Do we want to keep this?

Reply:
I kept it because we may want to revisit that alternative implementation in the future. Currently, we don't compute full rdwpc and zdwpc matrices (because that code was really slow), but we will probably need to eventually.